make.js

```diff
@@ -121,18 +90,36 @@ target.build = function() {
                 var modOutDir = path.join(commonOutputPath, modName);
 
                 if (!test('-d', modOutDir)) {
+                    banner('Building module ' + commonPath, true);
+
+                    // build the common node module
                     if (mod.type === 'node' && mod.compile == true) {
                         buildNodeTask(commonPath, modOutDir);
-                        copyTaskResources(commonPath, modOutDir);
-
-                        // install the common module to the task dir
-                        mkdir('-p', path.join(taskPath, 'node_modules'));
-                        rm('-Rf', path.join(taskPath, 'node_modules', modName));
-                        pushd(taskPath);
-                        run('npm install ' + modOutDir);
-                        popd();
+                    }
+
+                    // todo: copy additional resources based on the local make.json
+                    copyTaskResources(commonPath, modOutDir);
+
+                    // get externals
+                    var modMakePath = path.join(commonPath, 'make.json');
+                    var modMake = test('-f', modMakePath) ? require(modMakePath) : {};
+                    if (modMake.hasOwnProperty('externals')) {
+                        console.log('Getting module externals');
+                        getExternals(modMake.externals, modOutDir);
                     }
                 }
+
+                // npm install the common module to the task dir
+                if (mod.type === 'node' && mod.compile == true) {
+                    mkdir('-p', path.join(taskPath, 'node_modules'));
+                    rm('-Rf', path.join(taskPath, 'node_modules', modName));
```

Comment:
this `rm` command looks like a bug. this will whack any node_modules already installed in the task directory.